arviz/plots/hdiplot.py

```diff
@@ -109,6 +109,17 @@ def plot_hdi(
         >>> ax = az.plot_hdi(x_data, hdi_data=hdi_data[0], color="r", fill_kwargs={"alpha": .2})
         >>> az.plot_hdi(x_data, hdi_data=hdi_data[1], color="k", ax=ax, fill_kwargs={"alpha": .2})
 
+    ``plot_hdi`` can also be used with Inference Data objects. Here we use the posterior predictive
+    to plot the HDI interval.
+
+    .. plot::
+        :context: close-figs
+
+        >>> data = az.load_arviz_data('radon')
+        >>> x_data = np.array(data.posterior_predictive.obs_id)
+        >>> y_data = np.array(data.posterior_predictive.y)
+        >>> az.plot_hdi(x_data, y_data)
```

Comment:
not sure if you have looked at how the plot looks like, but `obs_id` are integer indexes that don't really have any meaning, they should not be used as x values. They are also not continuous, so plotting a continuous hdi band over all observations is not a good idea either. 

I think this should be done with the regression1d example better, but it looks like the InferenceData does not have the x values used as predictiors. Going down this path would require solving https://github.com/arviz-devs/arviz/issues/1291 and extending the code to get the x data into the inferencedata, which will probably take some time.

As a more immediate solution, I would use simulated data (maybe even the same as above) to create a new inferencedata instead of loading a preexisting one and then use this one to plot the hdi

Reply:
Yeah, that makes sense, I too was confused about this bit. I was also thinking of creating a new Inference Object and using that to plot the HDI but as the issue suggested to use pre-computed data so went on with that approach to see if it was possible. So as of now, it is not possible as there are no values in the posterior predictive or the posterior. Will make the necessary changes. Thanks

Reply:
Minor note, the x values would be in the `constant_data` group

Reply:
Done!, could you check @OriolAbril? Thanks